support streaming AGG and some fix in UDAF

stdpain opened the pull request:
## What type of PR is this：
- [x] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：
for #2742 

## Problem Summary(Required) ：
1. return error when exception happened in UDF
2. fix wrong result in UDAF group by
3. fix crash in scalar UDF when args has const column
4. support streaming aggregate when using UDAF

kangkaisen commented on be/src/exec/vectorized/aggregator.cpp:
```diff
@@ -532,6 +537,15 @@ void Aggregator::try_convert_to_two_level_set() {
     }
 }
 
+Status Aggregator::check_function_ctxs() {
```
check_has_error?

kangkaisen commented on be/src/exprs/expr_context.h:
```diff
@@ -165,4 +167,11 @@ class ExprContext {
     std::atomic<bool> _closed;
 };
 
+#define RETURN_IF_EVALUTE_ERROR(expr_ctxs)         \
```
```suggestion
#define RETURN_IF_HAS_ERROR(expr_ctxs)         \
```

stdpain committed:
support streaming AGG and some fix in UDAF

1. return error when exception happened in UDF
2. fix wrong result in UDAF group by
3. fix crash in scalar UDF when args has const column
4. support streaming aggregate when using UDAF

stdpain commented:
run starrocks_be_unittest

stdpain commented:
run starrocks_fe_unittest

stdpain commented:
run starrocks_fe_unittest
